src/models/room.js

```diff
@@ -1141,6 +1136,25 @@ Room.prototype.addPendingEvent = function(event, txnId) {
 
     this.emit("Room.localEchoUpdated", event, this, null, null);
 };
+// live events are aggregated in the timelineset
+// but for local echo (and undoing the local echo of a redaction) we do it here.
```

Comment:
I don't think I am following this line very well... Maybe something like:

```suggestion
// Since we want local echoes of relations (and to undo the local echo of redacted 
// relations), we aggregate them into the timeline set's relation storage instead of 
// keeping them detached as we do for regular messages.
```

Reply:
Please let me know if it's still hard to follow after the changes I made.

Reply:
It's much better now, thanks! 😁